homeassistant/components/sensor/islamic_prayer_times.py

```diff
@@ -0,0 +1,235 @@
+                                     date=str(today))
+
+        self.prayer_times = calc.fetch_prayer_times()
+        return self.prayer_times
+
+
+class IslamicPrayerTimeSensor(Entity):
+    """Representation of an Islamic prayer time sensor."""
+
+    ENTITY_ID_FORMAT = 'sensor.islamic_prayer_time_{}'
+
+    def __init__(self, sensor_type, prayer_times_data):
+        """Initialize the Islamic prayer time sensor."""
+        self.sensor_type = sensor_type
+        self.entity_id = self.ENTITY_ID_FORMAT.format(self.sensor_type)
+        self.prayer_times_data = prayer_times_data
+        self._display_format = "%I:%M%p"
+        self._name = self.sensor_type.capitalize()
+        prayer_time = self.prayer_times_data.prayer_times[self._name]
+        pt_dt = self.get_prayer_time_as_dt(prayer_time)
+        self._state = pt_dt.strftime(self._display_format)
+        _LOGGER.debug("%s State: %s", self.sensor_type, self._state)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to display in the front end."""
+        return PRAYER_TIMES_ICON
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Disable polling."""
+        return False
+
+    async def async_fire_prayer_event(self):
+        """Fire event for respective prayer time."""
+        _LOGGER.debug("Firing Event for: %s", self.sensor_type)
+        self.hass.bus.async_fire('islamic_prayer_time', {'prayer':
+                                                         self.sensor_type})
+
+    @staticmethod
+    def get_prayer_time_as_dt(prayer_time):
+        """Create a datetime object for the respective prayer time."""
+        today = datetime.today().strftime('%Y-%m-%d')
+        date_time_str = '{} {}'.format(str(today), prayer_time)
+        pt_dt = dt_util.parse_datetime(date_time_str)
+        return pt_dt
+
+    async def async_set_event_trigger(self):
+        """Set a trigger for when to fire an event."""
+        today = datetime.today().strftime('%Y-%m-%d')
+        date_time_str = '{} {}'.format(str(today), self._state)
+        trigger_time = datetime.strptime(date_time_str, '%Y-%m-%d %I:%M%p')
+
+        if datetime.now() < trigger_time:
+            # Only create an event for prayers times in the future
+            _LOGGER.debug("Creating event trigger for: %s", self._name)
+
+            async_track_point_in_time(self.hass,
+                                      self.async_fire_prayer_event(),
+                                      trigger_time)
+
+    async def async_update(self):
+        """Update the sensor."""
+        prayer_time = self.prayer_times_data.prayer_times[self.name]
+        pt_dt = self.get_prayer_time_as_dt(prayer_time)
+        self._state = pt_dt.strftime(self._display_format)
```

Comment:
state should be a iso formatted datetime stamp. Then the device class should be set to "timestamp".

Reply:
good point, i'll update this.